### PR TITLE
Add PayPalRequest Object

### PR DESCRIPTION
**Pull request by sarahkoop**
### Reason for changes

We have a `CardRequest` object for configuring parameters for card payments, we should follow this pattern for PayPal and all other features.

### Summary of changes

- Add `PayPalRequest` and update `PayPalClient.start` to accept a `PayPalRequest`

### Checklist

- ~[ ] Added a changelog entry~

- @jaxdesmarais 
- @sarahkoop

### CODE REVIEW

**Review by sshropshire — APPROVED** [review]
🌊

**Comment by Asavarkhul** [comment]
Hey there!
Nice Idea to follow the `CardRequest ` pattern 👌

I got a question, why using a struct and not a protocol to define what should be a `PayPalRequest `?

Thanks 🙏

File: Sources/PayPal/Models/PayPalRequest.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+/// Used to configure options for approving a PayPal order
+public struct PayPalRequest {
```
Comment (Asavarkhul) [inline Sources/PayPal/Models/PayPalRequest.swift:4]: Hey there!
Nice Idea to follow the `CardRequest` pattern 👌

I got a question, why using a struct and not a protocol to define what should be a `PayPalRequest` ?

Thanks 🙇‍♂️

Comment (jaxdesmarais) [reply Sources/PayPal/Models/PayPalRequest.swift:4]: While we could certainly use a protocol for the request and have the struct conform to it, the lack of reuse (at least currently) for the protocol as well as the current simplicity of the objects have lead us to use structs for the request objects. This also is a similar pattern to what Apple uses when constructing a `PKPaymentRequest` for Apple Pay. That said, if we see a strong need to move in a protocol direction in the future we would certainly explore this option.

Comment (Asavarkhul) [reply Sources/PayPal/Models/PayPalRequest.swift:4]: Thanks a lot @jaxdesmarais for your answer 🙏

**Review by jaxdesmarais — APPROVED** [review]
⛵  looks good!

**Review by notoriousmarcos — APPROVED** [review]

**Pull request merged**